ddsrouter_core/include/ddsrouter_core/types/endpoint/Endpoint.hpp

```diff
@@ -29,12 +29,21 @@ namespace ddsrouter {
 namespace core {
 namespace types {
 
+using EndpointKindType = unsigned int;
+
 //! Possible kinds of the endpoint
-enum EndpointKind
+enum class EndpointKind : EndpointKindType
+{
+    invalid = 0,
+    writer = 1,
+    reader = 2,
+};
+
+constexpr std::array<const char*, 3> EndpointKindStrings =
```

Comment:
**In general**

Do not use "magic numbers" along the code. I understand the needed of this value, but it should come from a macro or a value that may not change when the enumeration changes

Reply:
If this will remain like that (and maybe fix it in future PRs) it should have a TODO comment and a NOTE/WARNING comment in the enumeration itself reminding anyone who modify it that should modify this value